File: app/src/main/java/com/example/kotlinmvp/ui/fragment/CollapsingToolbarFragment.java
package com.example.kotlinmvp.ui.fragment;

import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.camming.mvp.ui.fragment.BaseFragment;
import com.example.kotlinmvp.R;
import com.google.android.material.appbar.AppBarLayout;
import com.google.android.material.appbar.CollapsingToolbarLayout;

import qiu.niorgai.StatusBarCompat;


/**
 * Compat CollapsingToolbarLayout
 * Created by qiu on 7/27/16.
 */
public class CollapsingToolbarFragment extends BaseFragment {

    public CollapsingToolbarFragment() {

    }

    private AppBarLayout mAppBarLayout;
    private CollapsingToolbarLayout mCollapsingToolbarLayout;
    private Toolbar mToolbar;


    @Override
    public int getFragmentLayoutId() {
        return R.layout.fragment_collapsing_tool_bar;
    }

    @Override
    public void initView() {

    }

    @Override
    public void initData() {

    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setHasOptionsMenu(true);

        mAppBarLayout = (AppBarLayout) view.findViewById(R.id.appbar);
        mCollapsingToolbarLayout = (CollapsingToolbarLayout) view.findViewById(R.id.collapsing_toolbar);
        mCollapsingToolbarLayout.setTitle(getString(R.string.app_name));
        mToolbar = (Toolbar) view.findViewById(R.id.toolbar);
        ((AppCompatActivity) getActivity()).setSupportActionBar(mToolbar);
        ((AppCompatActivity) getActivity()).getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        StatusBarCompat.setStatusBarColorForCollapsingToolbar(getActivity(), mAppBarLayout, mCollapsingToolbarLayout, mToolbar, Color.TRANSPARENT);
    }

    @Override
    public void onHiddenChanged(boolean hidden) {
        super.onHiddenChanged(hidden);
        if (!hidden) {
            StatusBarCompat.setStatusBarColorForCollapsingToolbar(getActivity(), mAppBarLayout, mCollapsingToolbarLayout, mToolbar, Color.TRANSPARENT);
        }
    }
}

File: app/src/main/java/com/example/kotlinmvp/ui/fragment/CollapsingToolbarFragment.kt
package com.example.kotlinmvp.ui.fragment

import android.graphics.Color
import com.camming.mvp.ui.fragment.BaseFragment
import com.example.kotlinmvp.R
import kotlinx.android.synthetic.main.fragment_collapsing_tool_bar.*
import qiu.niorgai.StatusBarCompat


class CollapsingToolbarFragment : BaseFragment() {
//    private var mAppBarLayout: AppBarLayout? = null
//    private var mCollapsingToolbarLayout: CollapsingToolbarLayout? = null
//    private var mToolbar: Toolbar? = null
    override fun getLayoutId(): Int {
        return R.layout.fragment_collapsing_tool_bar
    }
    override fun initListener() {

    }
    override fun initView() {
        //        ((AppCompatActivity) getActivity()).setSupportActionBar(mToolbar);
//        ((AppCompatActivity) getActivity()).getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        StatusBarCompat.setStatusBarColorForCollapsingToolbar(
            activity!!,
            appbar,
            collapsing_toolbar,
            toolbar,
            Color.TRANSPARENT
        )
    }
    override fun initData() {}

    override fun onHiddenChanged(hidden: Boolean) {
        super.onHiddenChanged(hidden)
        if (!hidden) {
            StatusBarCompat.setStatusBarColorForCollapsingToolbar(
                activity!!,
                appbar,
                collapsing_toolbar,
                toolbar,
                Color.TRANSPARENT
            )
        }
    }
}